libs/blocks/global-navigation/global-navigation.js

```diff
@@ -505,8 +505,46 @@ class Gnav {
     };
 
     const onAnalyticsEvent = (data) => {
-      // TODO: update based on requirements
-      console.log('analytics data', data);
+      if (!data) return;
+
+      const getInteraction = () => {
+        const { event: { type, subtype } = {}, source: { name } = {} } = data;
+        const contentName = data.content?.name;
+        switch (`${name}|${type}|${subtype}|${contentName || ''}`) {
+          case 'profile|click|sign-in|':
+            return `Sign in|gnav|${getExperienceName()}|unav`;
+          case 'profile|render|component|':
+            return `Account|gnav|${getExperienceName()}`;
+          case 'profile|click|account|':
+            return `View Account|gnav|${getExperienceName()}`;
+          case 'profile|click|sign-out|':
+            return `Sign out|gnav|${getExperienceName()}|unav`;
+          case 'app-switcher|render|component|':
+            return 'AppLauncher.appIconToggle';
+          case `app-switcher|click|app|${contentName}`:
+            return `AppLauncher.appClick.${contentName?.split('-')
```

Comment:
I'd still put this in a separate function
```
function convertToPascalCase(str) {
  return str
    .split('-')
    .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
    .join(' ');
}
...
app: `AppLauncher.appClick.${convertToPascalCase(data.content?.name || '')}`,
```

Reply:
Done 👍 Moved it to a separate function. Thanks for the suggestion 👍